### PR TITLE
Maintain E2E image Dockerfile

### PR DESCRIPTION
**Pull request by danfengliu**
E2E image Dockerfile has lost control for several updates, so o add modifications back to the Dockerfile. 1. update  and docker-compose; 
2. Install helm and helm3.
3. dockerd-entrypoint.sh has been modify by commit image manually, so it should be updated.

Signed-off-by: Danfeng Liu (c) <danfengl@vmware.com>

### CODE REVIEW

File: tests/e2e-image/dockerd-entrypoint.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ fi
 if [ "$1" = 'dockerd' ]; then
 	# if we're running Docker, let's pipe through dind
 	# (and we'll run dind explicitly with "sh" since its shebang is /bin/bash)
-	set -- sh "$(which dind)" "$@" "--insecure-registry=$IP"
+	set -- sh "$(which dind)" "$@" "--insecure-registry=0.0.0.0/0"
```
Comment (reasonerjt) [inline tests/e2e-image/dockerd-entrypoint.sh]: may I know why we need this change?  Just make sure this is not a shortcut which will cause problem in future...

File: tests/e2e-image/Dockerfile [hunk 2]
```diff
@@ -62,8 +76,16 @@ RUN wget https://github.com/drone/drone-cli/releases/download/v0.8.3/drone_linux
 RUN curl -sSL https://github.com/vmware/govmomi/releases/download/v0.16.0/govc_linux_amd64.gz | gzip -d > /usr/local/bin/govc && \
     chmod +x /usr/local/bin/govc
 
-RUN  wget https://launchpad.net/ubuntu/+source/wget/1.18-2ubuntu1/+build/10470166/+files/wget_1.18-2ubuntu1_amd64.deb && \
-     dpkg -i wget_1.18-2ubuntu1_amd64.deb
+RUN wget https://get.helm.sh/helm-v2.14.1-linux-386.tar.gz && tar zxvf helm-v2.14.1-linux-386.tar.gz && \
+    mv linux-386/helm /usr/local/bin/helm && \
+	helm init --client-only && \
+	helm plugin install https://github.com/chartmuseum/helm-push
```
Comment (reasonerjt) [inline tests/e2e-image/Dockerfile]: can we pin to a certain version of push plugin?

Comment (danfengliu) [reply tests/e2e-image/Dockerfile]: helm3 plugin install https://github.com/chartmuseum/helm-push

Currently, we have both helm push and helm3 push commands.

Comment (danfengliu) [reply tests/e2e-image/Dockerfile]: Do you mean pin helm to helm2? Currently, helm's verson is 2, helm3's version is 3.

**Comment by reasonerjt** [comment]
Please refine the commit message

Make sure there's an empty line between the title and actual message.  Look at other commit in this repo.

**Pull request merged**